### ISSUE TITLE
Point-in-polygon error for large polygon

### ISSUE THREAD

**Issue by rubak** [spatstat/spatstat#94]
Question on SO: <https://stackoverflow.com/q/54592460/3341769>

Minimal reproducible example:

```r
W1 <- owin(poly=1e6*cbind(c(-1,1,1,-1), c(-1,-1,1,1)))
inside.owin(0,0,W) # FALSE
W2 <- owin(poly=1e5*cbind(c(-1,1,1,-1), c(-1,-1,1,1)))
inside.owin(0,0,W2) # TRUE
```

Can we detect when the polygons are too large for point-in-polygon test to be reliable?

**Comment by baddstats** [comment]
Thanks. Especially for the MWE!
Your reply on stackoverflow said that we convert everything to integer coordinates; but this is only true of the more recent polyclip code. The rest of spatstat and spatstat.utils uses double precision coordinates. In this case the offending code is the ancient spatstat.utils::inside.xypolygon and its associated C routines. I will see if this is fixable...!

A

Prof Adrian Baddeley DSc FAA
John Curtin Distinguished Professor
Department of Mathematics and Statistics
Curtin University, Perth, Western Australia
________________________________

**Comment by baddstats** [comment]
This bug occurs only when the window is a large polygon (`type="polygonal"`) that is geometrically equivalent to a rectangle, as in the examples above. 

Large windows are rescaled by `inside.owin` to avoid numerical errors. The rescaling was done by `rescale.owin -> affine.owin` which had the default behaviour `rescue=TRUE` meaning that a polygon equivalent to a rectangle would be replaced by a rectangle (window of `type="rectangle"`). The changed structure was unexpected in `inside.owin` and caused the erroneous results.

It is now fixed in development version `spatstat 1.58-2.026`

I have also changed the behaviour of `rescale.owin` so that the geometrical type of the window is preserved.
